### PR TITLE
[ETCM-456] Properly display and handle sync status independently from…

### PR DESCRIPTION
**Pull request by kapke**
… wallet state

### CODE REVIEW

**Review by conqeror — APPROVED** [review]
I'm happy that the whole state polling is getting more reasonable step by step :) Tested and works well!

File: src/common/wallet-state.ts [hunk 1]
```diff
@@ -728,6 +734,11 @@ function useWalletState(initialState?: Partial<WalletStateParams>): WalletData {
     }
   }
 
+  useRecurringTimeout(async () => {
+    await refreshSyncStatus()
+    rendererLog.debug(`sync status: ${syncStatusOption}`)
+  }, 3000)
+
   return {
```
Comment (conqeror) [inline src/common/wallet-state.ts:742]: One comment actually, I would not export `refreshSyncStatus` from wallet state now. It should be responsibility of the wallet state to update this sync status, right?

Comment (kapke) [reply src/common/wallet-state.ts:742]: You're right! Fixing

File: src/common/wallet-state.ts [hunk 2]
```diff
@@ -728,14 +730,18 @@ function useWalletState(initialState?: Partial<WalletStateParams>): WalletData {
     }
   }
 
+  useRecurringTimeout(async () => {
+    await refreshSyncStatus()
+    rendererLog.debug(`sync status: ${syncStatusOption}`)
```
Comment (qwhex) [inline src/common/wallet-state.ts:735]: Only logs: `sync status: [object Object]`

File: src/wallets/Wallets.tsx [hunk 3]
```diff
@@ -1,55 +1,30 @@
...
 
 export const Wallets = (): JSX.Element => {
   const walletState = WalletState.useContainer()
 
-  useEffect(() => {
-    if (walletState.walletStatus === 'INITIAL') {
-      walletState.refreshSyncStatus()
-    }
-
-    rendererLog.debug(`walletStatus: ${walletState.walletStatus}`)
```
Comment (qwhex) [inline src/wallets/Wallets.tsx:21]: It was pretty useful to have this as a debug log. Could you add it back somewhere?

**Review by qwhex — APPROVED** [review]
I welcome these changes and from my testing, it behaves well.
Only small issues.

File: src/common/wallet-state.ts [hunk 4]
```diff
@@ -303,7 +304,7 @@ function useWalletState(initialState?: Partial<WalletStateParams>): WalletData {
   const syncStatus = getOrElse(
     (): SynchronizationStatus => ({
       mode: 'offline',
-      currentBlock: -1,
+      currentBlock: 0,
```
Comment (qwhex) [inline src/common/wallet-state.ts:307]: Wallet doesn't load if the fetched sync status is offline with currentBlock == 0.
It was -1 to make sure it gets updated by every possible fetched sync status.
Feel free to change the logic in `refreshSyncStatus` if you don't like having -1 here.

**Review by qwhex — CHANGES_REQUESTED** [review]

File: src/common/wallet-state.ts (hunk 4 above)
Comment (kapke) [reply src/common/wallet-state.ts:313]: Fixed

File: src/wallets/Wallets.tsx (hunk 3 above)
Comment (qwhex) [reply src/wallets/Wallets.tsx:21]: It logs every time when sync status is updated, not when walletStatus changed, so it doesn't help with tracking status changes.

**Review by qwhex — APPROVED** [review]
LGTM

**Pull request merged**